File: frontend/src/services/event.service.js
import axios from "axios";

const API_URL = "http://localhost:3001/api";

export const createEvent = async (eventData) => {
    try {
        const response = await axios.post(`${API_URL}/events`, eventData);
        return response.data;
    } catch (error) {
        throw error.response?.data || error;
    }
};

//export const fetchEvents = async (filters = {}) => {
    //const queryString = new URLSearchParams(filters).toString();
   // try {
        //const response = await axios.get(`${API_URL}/events?${queryString}`);
      //  return response.data;
    //} catch (error) {
  //      throw error.response?.data || error;
//    }
//};

export const fetchReadyEvents = async (swimmerId) => {
    try {
        console.log("Fetching events for swimmer_id:", swimmerId); // Debugging log
        const response = await axios.get(`${API_URL}/events-ready`, {
            params: { swimmer_id: swimmerId }, // Pass swimmer_id as a query parameter
        });
        console.log("API Response:", response.data); // Debugging log
        return response.data;
    } catch (error) {
        console.error("API Error:", error.response?.data || error); // Debugging log
        throw error.response?.data || error;
    }
};


export const attendEvent = async (swimmerId, eventId) => {
    try {
        const response = await axios.post(`${API_URL}/events/attend`, {
            swimmer_id: swimmerId,
            event_id: eventId,
        });
        return response.data;
    } catch (error) {
        throw error.response?.data || error;
    }
};

export const cancelEvent = async (eventData) => {
    try {
        const response = await axios.post(
            `${API_URL}/cancel-event`,
            eventData,
            {
                headers: {
                    Authorization: `Bearer ${localStorage.getItem('token')}`,
                },
            }
        );
        return response.data;
    } catch (error) {
        throw error.response?.data || error;
    }
};

export const fetchAllReadyEvents = async () => {
    try {
        const response = await axios.get(`${API_URL}/all-ready-events`);
        return response.data;
    } catch (error) {
        throw error.response?.data || error;
    }
};

export const fetchEventTypes = async () => {
    try {
        const response = await axios.get(`${API_URL}/event-types`);
        return response.data;
    } catch (error) {
        throw error.response?.data || error;
    }
};

File: backend/services/event_service.py
from database.queries.event_queries import create_event, fetch_manager_pools
from database.queries.event_queries import fetch_ready_events, add_attendance, cancel_event
from database.queries.event_queries import fetch_all_ready_events, fetch_event_types

def create_new_event(event_data, manager_id):
    try:
        create_event(event_data, manager_id)
    except Exception as e:
        raise Exception(f"Error in service while creating event: {e}")

def get_manager_pools(manager_id):
    try:
        return fetch_manager_pools(manager_id)
    except Exception as e:
        raise Exception(f"Error in service while fetching pools: {e}")
def get_ready_events(swimmer_id):
    try:
        return fetch_ready_events(swimmer_id)
    except Exception as e:
        raise Exception(f"Error in service while fetching ready events: {e}")



def register_to_event(swimmer_id, event_id):
    try:
        add_attendance(swimmer_id, event_id)
    except Exception as e:
        raise Exception(f"Error in service while adding attendance: {e}")
    
def cancel_event_by_id(event_id):
    cancel_event(event_id)

def get_all_ready_events():
    try:
        return fetch_all_ready_events()
    except Exception as e:
        raise Exception(f"Error in service while fetching all ready events: {e}")

def get_event_types():
    try:
        return fetch_event_types()
    except Exception as e:
        raise Exception(f"Error in service while fetching event types: {e}")